File: materials-evaluation-dotnet/Database/Models/MaterialBatch.cs
namespace MaterialsEvaluation.Database
{
    public class MaterialBatch
    {
        public Guid Id { get; set; }

        public Guid MaterialId { get; set; }
        public virtual Material Material { get; set; }

        public Guid QualityVisionId { get; set; }
        public virtual QualityVision QualityVision { get; set; }

        public DateTime CreatedAt { get; set; }
        public int AmountOfTests { get; set; }
        public DateTime? CalculatedAt { get; set; }
        public string Status { get; set; }

        public List<MaterialBatchTests> MaterialBatchTests { get; set; }

        public MaterialBatch(
            Guid id,
            Guid materialId,
            Guid qualityVisionId,
            DateTime createdAt,
            int amountOfTests,
            DateTime? calculatedAt,
            string status
        )
        {
            Id = id;
            MaterialId = materialId;
            QualityVisionId = qualityVisionId;
            CreatedAt = createdAt;
            AmountOfTests = amountOfTests;
            CalculatedAt = calculatedAt;
            Status = status;
        }
    }
}

File: materials-evaluation-spa/src/app/models/MaterialBatch.ts
import { Material } from './Material';
import { QualityProperty } from './QualityProperty';
import { QualityVision } from './QualityVision';

export enum Status {
  Pending,
  OutOfRange,
  InRange,
}

export interface Test {
  qualityProperty: QualityProperty;
  createdAt: string;
  resultQualitative: boolean | null;
  resultQuantitative: number | null;
}

export interface MaterialBatch {
  id: string;
  material: Material;
  qualityVision: QualityVision;
  createdAt: string;
  amountOfTests: number;
  calculatedAt: string | null;
  status: Status;
  tests: Test[];
}
